### PR TITLE
[Config] Add tests to compare generated code

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Adding tests that compare the generated code has several benefits
- assert that the PR does not break the generated code (that should be part of our BC promises)
- easier to review the PR that touch the CodeGenerator

This PR adds nothing but initializing the Fixtures without output generated by the current implementation.

The single file changed is src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php

### CODE REVIEW

**Comment by GromNaN** [comment]
I like snapshot testing for the reasons you mentioned. We are supposed to update the generated code manually when a change is done? Adding an flag (env var) to force update would be very useful. 

For reference, I use this library for snapshot testing: https://github.com/spatie/phpunit-snapshot-assertions

File: src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php [hunk 1]
```diff
@@ -48,12 +49,16 @@ public function fixtureNames()
      */
     public function testConfig(string $name, string $alias)
     {
+        $outputDir = sys_get_temp_dir().\DIRECTORY_SEPARATOR.uniqid('sf_config_builder', true);
```
Comment (GromNaN) [inline src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php]: What about deleting this directory on `tearDown`?

Comment (jderusse) [reply src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php]: done

**Comment by jderusse** [comment]
Not sure it worth a flag, but I added a commented snippet to help people generate code

File: src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php [hunk 2]
```diff
@@ -128,13 +160,31 @@ private function generateConfigBuilder(string $configurationClass)
...
+        foreach (new \RecursiveIteratorIterator(new RecursiveDirectoryIterator($current, \FilesystemIterator::SKIP_DOTS)) as $file) {
+            if ($file->isDir()) {
+                continue;
+            }
+            $currentFiles[substr($file->getPathname(), \strlen($current))] = $file->getPathname();
+        }
+
+        $this->assertSame(array_keys($expectedFiles), array_keys($currentFiles));
+        foreach ($expectedFiles as $fileName => $filePath) {
+            $this->assertSame(file_get_contents($filePath), file_get_contents($currentFiles[$fileName]));
```
Comment (HypeMC) [inline src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php]: Couldn't `assertFileEquals` be used here instead?

```suggestion
            $this->assertFileEquals($filePath, $currentFiles[$fileName]);
```

Comment (jderusse) [reply src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php]: I didn't know that method exists. Thanks :heart:

**Review by GromNaN — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by GromNaN** [comment]
Thank you @jderusse.

**Comment by nicolas-grekas** [comment]
The test failures on appveyor where not false positives, this should have been fixed before merging.
@jderusse can you please have a look?

**Comment by stof** [comment]
Looks like some part of the code generator using `PHP_EOL` instead of `"\n"`...

**Comment by jderusse** [comment]
> The test failures on appveyor where not false positives, this should have been fixed before merging. @jderusse can you please have a look?

followup #44191

**Pull request closed without merging**